### PR TITLE
Mark generated files as generated in .gitattributes

### PR DESCRIPTION
**Pull request by Wumpf**
### What
... which implies .gitattributes that are itself generated!

Also (offtopic), shared `write_file` utility accross generators.

### Checklist
* [x] I have read and agree to [Contributor Guide](https://github.com/rerun-io/rerun/blob/main/CONTRIBUTING.md) and the [Code of Conduct](https://github.com/rerun-io/rerun/blob/main/CODE_OF_CONDUCT.md)
* [x] I've included a screenshot or gif (if applicable)
* [x] I have tested [demo.rerun.io](https://demo.rerun.io/pr/2777) (if applicable)

- [PR Build Summary](https://build.rerun.io/pr/2777)
- [Docs preview](https://rerun.io/preview/pr%3Aandreas%2Fcodegen%2Fmark-files-as-generated/docs)
- [Examples preview](https://rerun.io/preview/pr%3Aandreas%2Fcodegen%2Fmark-files-as-generated/examples)

### CODE REVIEW

File: crates/re_types/.gitattributes [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+src/datatypes/quaternion.rs linguist-generated=true
+src/datatypes/rotation3d.rs linguist-generated=true
+src/datatypes/rotation_axis_angle.rs linguist-generated=true
+src/datatypes/scale3d.rs linguist-generated=true
+src/datatypes/transform3d.rs linguist-generated=true
+src/datatypes/translation_and_mat3x3.rs linguist-generated=true
+src/datatypes/translation_rotation_scale3d.rs linguist-generated=true
+src/datatypes/vec2d.rs linguist-generated=true
+src/datatypes/vec3d.rs linguist-generated=true
+src/datatypes/vec4d.rs
```
Comment (teh-cmc) [inline crates/re_types/.gitattributes:35]: why is this guy different? :disappointed:

Comment (Wumpf) [reply crates/re_types/.gitattributes:35]: huh. good catch. that's because `join(..)` is not what I wnated here!

Comment (Wumpf) [reply crates/re_types/.gitattributes]: fixed

File: crates/re_types_builder/src/lib.rs [hunk 2]
```diff
@@ -253,6 +253,29 @@ pub fn generate_lang_agnostic(
...
+        std::iter::once(filename.to_owned()) // The attributes itself is generated!
+            .chain(files.map(|path| {
+                path.strip_prefix(output_path.as_ref().as_std_path())
+                    .context("Failed to make path relative to output path.")
+                    .unwrap()
+                    .to_string()
+            }))
+            .map(|s| format!("{s} linguist-generated=true\n"))
+            .collect::<Vec<_>>()
+            .join(""),
```
Comment (Wumpf) [inline crates/re_types_builder/src/lib.rs:275]: not putting \n here because I want trailing newline

**Review by teh-cmc — APPROVED** [review]

**Pull request merged**